packages/aws-cdk/lib/api/util/sdk.ts

```diff
@@ -290,4 +331,62 @@ function httpsProxyFromEnvironment(): string | undefined {
         return process.env.HTTPS_PROXY;
     }
     return undefined;
-}
\ No newline at end of file
+}
+
+/**
+ * Return whether it looks like we'll have ECS credentials available
+ */
+function hasEcsCredentials() {
+    return (AWS.ECSCredentials.prototype as any).isConfiguredForEcsCredentials();
+}
+
+/**
+ * Return whether we're on an EC2 instance
+ */
+async function hasEc2Credentials() {
```

Comment:
Surely, this is not something that could be included in all the SDKs, right? 🤨

Reply:
Probably most other ones have semi-sane behavior in this regard already.